src/blockfilter.cpp

```diff
@@ -52,6 +52,9 @@ static uint64_t GolombRiceDecode(BitStreamReader<IStream>& bitreader, uint8_t k)
 // See: https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction/
 static uint64_t MapIntoRange(uint64_t x, uint64_t n)
 {
+#ifdef __SIZE_INT128__
```

Comment:
This doesn't seem to work for me (x86_64 linux, GCC 7.3.0), verified by looking at the disassembled code.

Interestingly, there is hardly any impact on the benchmarks when I modify the code to use force using 128-bit multiplication - suggesting that most of the time is spent elsewhere (hashing, bit operations?).

Reply:
Oops, that's a typo. It should be `__SIZEOF_INT128__`. Will fix.

Reply:
Yup, that works.